node/src/components/block_executor.rs

```diff
@@ -458,6 +480,16 @@ impl BlockExecutor {
                 .map(|summary| summary.state_root_hash)
         }
     }
+
+    /// Returns true if the `finalized_block` is an immediate child of the initial block, ie.
+    /// either genesis or the highest known block at the time of initializing the component.
+    fn is_initial_block_child(&self, finalized_block: &FinalizedBlock) -> bool {
+        finalized_block.height()
+            == self
+                .initial_block_height
+                .map(|height| height + 1)
+                .unwrap_or(0)
```

Comment:
I don't think we should do this here; I suggest we set `initial_block_height` in the constructor.